examples/screen_counter.rs

```diff
@@ -67,37 +66,48 @@ fn main() -> ! {
 
     info!("On screen counter demo running!");
 
-    // TODO: Find a way to make the buzzer work again.
-
     // Emit a sound
-    //buzzer.set_frequency(Rate::from_hz(261));
-    //let mut delay = Delay::new();
-    //delay.delay_ms(100);
-    //buzzer.off();
+    let mut tone_freq: u32 = 261;
+    let mut buzzer = buzzer
+        .tone(tone_freq as u16, 100)
+        .expect("start tone failed");
 
-    let mut pos: i32 = 1;
+    let mut pos: i32 = 0;
+    let mut pos_delta: i32;
     let mut need_redraw = true;
     loop {
         // Change Test color on button push
         if let Some(state) = board.has_button_changed() {
-            if state == false {
+            if !state {
                 style_index = (style_index + 1) % STYLE_LIST.len();
                 need_redraw = true;
             }
         }
 
         // Test if encoder has rotated
-        match encoder.update().unwrap() {
-            Direction::Clockwise => {
-                pos += 1;
-                need_redraw = true;
-            }
-            Direction::CounterClockwise => {
-                pos -= 1;
-                need_redraw = true;
-            }
-            Direction::None => {}
+        pos_delta = match encoder.update().unwrap() {
+            Direction::Clockwise => 1,
+            Direction::CounterClockwise => -1,
+            Direction::None => 0,
+        };
+
+        if pos_delta < 0 {
+            need_redraw = true;
+            tone_freq = (tone_freq * 1000) / 1059;
```

Comment:
Just out of curiosity, where this 1059 comes from ?

Reply:
Oh! Sorry about that. 1059 is `round(2^(1/12) * 1000)` so 12 multiplications will double the frequency. It's not perfectly tuned but it's close. Ideally, the synth would have at least the option to include a precomputed LUT for most 12-tone notes between 20 Hz and 20 kHz.

Reply:
Ha, that's some music physics I gess. I know nothing in that field. Thanks for the explanation.